src/main/java/org/maveric/quarkus/panache/services/TransactionService.java

```diff
@@ -0,0 +1,93 @@
+package org.maveric.quarkus.panache.services;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.transaction.Transactional;
+import lombok.extern.slf4j.Slf4j;
+import org.maveric.quarkus.panache.dtos.TransactionRequestDto;
+import org.maveric.quarkus.panache.dtos.TransactionResponseDto;
+import org.maveric.quarkus.panache.enums.TransactionType;
+import org.maveric.quarkus.panache.exceptionHandler.InsufficientFundsException;
+import org.maveric.quarkus.panache.exceptionHandler.SavingsAccountDetailsNotFoundException;
+import org.maveric.quarkus.panache.model.SavingsAccount;
+import org.maveric.quarkus.panache.model.Transaction;
+import org.maveric.quarkus.panache.repository.SavingsAccountRepository;
+import org.maveric.quarkus.panache.repository.TransactionRepository;
+import org.modelmapper.ModelMapper;
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+@Slf4j
+public class TransactionService {
+    @Inject
+    private TransactionRepository transactionRepository;
+    @Inject
+    private SavingsAccountRepository savingAccountRepository;
+    @Inject
+    private ModelMapper mapper;
+
+    @Transactional
+    public boolean deposit(TransactionRequestDto requestDto) {
+        Long accountId = requestDto.getAccountId();
+        BigDecimal amount = requestDto.getAmount();
+        SavingsAccount account = savingAccountRepository.findBySavingsAccountId(requestDto.getAccountId());
+        if (account == null) {
+            throw new SavingsAccountDetailsNotFoundException("Account not found");
+        }
+        Transaction transaction = new Transaction();
+        transaction.setSavingAccount(account);
+        transaction.setAmount(amount);
+        transaction.setDescription("Bengaluru");
+        transaction.setDate(Instant.now());
+        transaction.setType(TransactionType.DEPOSIT);
+        transaction.setBalance(account.getBalance().add(amount));
+        transactionRepository.persist(transaction);
+        account.setBalance(transaction.getBalance());
+        savingAccountRepository.persist(account);
+        System.out.println("Executing");
```

Comment:
remove sysouts.

Reply:
dont return a boolean value. it can be a void. if the statement got completed successfully it will execute next step in controller. otherwise it will goto exception scenario

Reply:
Sorry i was checking that's why i putted sout there and i removed that one 
First i made it void then loan service developer said change it to boolean so made it to boolean , i will change it to void now